engine/src/main/java/org/terasology/rendering/nui/editor/NUIEditorScreen.java

```diff
@@ -281,22 +295,89 @@ private void paste(JsonTree item) {
+
+        inlineEditorEntry.bindValue(new Binding<JsonTree>() {
+            @Override
+            public JsonTree get() {
+                return item;
+            }
+
+            @Override
+            public void set(JsonTree value) {
+                if (value != null) {
+                    item.setValue(value.getValue());
+                    editorTreeView.clearAlternativeWidgets();
+                    updateTreeView(item.getRoot());
+                    updateWidget(item.getRoot());
+                }
+            }
+        });
+
+        inlineEditorEntry.setFormatter(value -> {
+            JsonObject jsonObject = new JsonObject();
+
+            String jsonKey = value.getValue().getKey();
+            Object jsonValue = value.getValue().getValue();
+
+            if (jsonValue instanceof Boolean) {
+                jsonObject.addProperty(jsonKey, (Boolean) jsonValue);
+            } else if (jsonValue instanceof Number) {
+                jsonObject.addProperty(jsonKey, (Number) jsonValue);
+            } else if (jsonValue instanceof String) {
+                jsonObject.addProperty(jsonKey, (String) jsonValue);
+            } else {
+                jsonObject.addProperty(jsonKey, (Character) jsonValue);
+            }
+
+            String jsonString = new Gson().toJson(jsonObject);
+            return jsonString.substring(1, jsonString.length() - 1);
+        });
+
+        inlineEditorEntry.setParser(value -> {
+            String jsonString = String.format("{%s}", value);
+            try {
+                JsonElement jsonElement = new JsonParser().parse(jsonString);
+                Map.Entry keyValuePair = jsonElement.getAsJsonObject().entrySet().iterator().next();
+
+                String jsonKey = (String) keyValuePair.getKey();
+                JsonTreeNode node;
+
+                if (keyValuePair.getValue() == null) {
+                    node = new JsonTreeNode(jsonKey, null, JsonTreeNode.ElementType.KEY_VALUE_PAIR);
+                } else {
+                    JsonPrimitive jsonValue = (JsonPrimitive) keyValuePair.getValue();
+                    if (jsonValue.isBoolean()) {
+                        node = new JsonTreeNode(jsonKey, jsonValue.getAsBoolean(), JsonTreeNode.ElementType.KEY_VALUE_PAIR);
+                    } else if (jsonValue.isNumber()) {
+                        node = new JsonTreeNode(jsonKey, jsonValue.getAsNumber(), JsonTreeNode.ElementType.KEY_VALUE_PAIR);
+                    } else {
+                        node = new JsonTreeNode(jsonKey, jsonValue.getAsString(), JsonTreeNode.ElementType.KEY_VALUE_PAIR);
+                    }
+                }
+                return new JsonTree(node);
+            } catch (JsonSyntaxException e) {
+                return null;
+            }
+        });
+
+        editorTreeView.clearAlternativeWidgets();
+        editorTreeView.addAlternativeWidget(editorTreeView.getModel().indexOf(item), inlineEditorEntry);
```

Comment:
I can't think of a use casd where we want multiple inline editors active at the same time.
So maybe a `showEditorWidgetForActiveItem(EditorWidget)` would be enough.
Or TreeView could offer a registerEditorForNodeType method.(Since usually double click and F2 activa the renaming/editing of an item and would thrn not to be implemented for each treeview useage again